extruct/microformat.py

```diff
@@ -0,0 +1,14 @@
+import mf2py
+
+class MicroformatExtractor(object):
+
+    def extract(self, htmlstring, url='http://www.example.com/', encoding='UTF-8'):
+        return list(self.extract_items(htmlstring, url=url))
+
+    def extract_items(self, html, url, document=None):
+        for obj in mf2py.parse(html, html_parser="lxml", url=url)['items']:
+            if obj['type'] == 'h-entry':
+                for o in obj['children']:
```

Comment:
As far as I understand, this will turn a nested structure into a flat one - is this something that we want to do? Any examples where this happens in practice?

Reply:
this page for example: https://shop.wwe.com/aj-styles-fastlane-2018-15-x-17-framed-plaque-w%2F-ring-canvas/CN1148.html?dwvar_CN1148_color=No%20Color#curr=USD&start=2
if you do 
```import requests
import mf2py

resp = requests.get(url, timeout=30)
mf2py.parse(resp.content, html_parser='lxml')['items'][10]```
This is not a product item but it's the structure it flattens. I could not find an example for a product with this structure but I guess they are out there. I imagine this was added to aid with making extracted content more uniform.

Reply:
@Kebniss maybe the page changed, I get this:
```
{'children': [{'properties': {'name': ['Buy One, Get One for $5']},
   'type': ['h-hidden-phone']},
  {'properties': {'name': ['*Offer Ends Mar 28 02:59 AM EDT']},
   'type': ['h-hidden-phone']}],
 'properties': {'name': ['Buy One, Get One for $5\n-SHOP NOW-\n\n\n*Offer Ends Mar 28 02:59 AM EDT\nView Details\n\n\n\nBuy One Tee, Get One for $5. No code necessary. Exclusions apply (Preorder/Backorder, WrestleMania 34, Print on Demand, Homage, Finn Balor OGBC T-Shirt, Clearance). Cannot be combined with any other offer.\nEnds Mar 28\xa002:59 AM EDT']},
 'type': ['h-hidden-phone']}
```
which does not have ``'h-entry'`` type. 

> I imagine this was added to aid with making extracted content more uniform.

I'm mostly worried that in some cases this might loose valuable information.

Reply:
That's correct, I just wanted to give you an example of the type of structure it flattens. I did not find any of these structures under an h-entry. 
We are losing valuable information because we are only returning `h-entry` objects. Microformat has entities specific for products that we should add. I mentioned this in issue #66.

Reply:
> We are losing valuable information because we are only returning h-entry objects. 

Did you mean "we are *not* loosing"? If yes, will it change if we start returning other objects besides ``h-entry``?

Reply:
My bad I read the code wrong. We are already returning all types of microformat and only flattening h-entry. It could make sense to simply return everything as is and then apply transformations in another library.